`test_math` is flaky

This is part of https://github.com/chainer/chainer/issues/6903 effort.

- Failure example: https://ci.preferred.jp/chainer.py37/2259/
- Failure test: chainerx_tests/unit_tests/routines_tests/test_math.py
- Failure test case: `test_Tan_param_17_{contiguous=None, in_dtypes=('float16',), out_dtype='float16', input=2, shape=(2, 3)}[native:0]`
- Tolerance:
```
E           Not equal to tolerance rtol=0.001, atol=0.001
E           
E           Mismatch: 100%
E           Max absolute difference: 0.00203618
E           Max relative difference: 0.00200944
```

It seems the test is still flaky in the master branch.

- Failure example: https://ci.preferred.jp/chainer.py37/2762/
- Failure test: chainerx_tests/unit_tests/routines_tests/test_math.py
- Failure test case: `test_LogSumExp_param_1_{keepdims=False}_param_1_{shape=(2,), axis=-1}_param_0_{in_dtypes=('float16',), out_dtype='float16'}[cuda:0]`
- Tolerance:
```
E       Not equal to tolerance rtol=0.001, atol=0.001
E       
E       Mismatch: 100%
E       Max absolute difference: 0.00116
E       Max relative difference: 0.011566
```

FYI, this is another example:

- Failure example: https://ci.preferred.jp/chainer.py37/2758/
- Failure test case: `test_Min_param_105_{in_dtypes=('float16',), out_dtype='float16', is_module=False, shape=(4, 2), axis=-2}[cuda:0]`
- Tolerance:
```
E           Not equal to tolerance rtol=0.01, atol=0.01
E           
E           Mismatch: 100%
E           Max absolute difference: 95.84712927
E           Max relative difference: 1179.51587459
```

--- a/tests/chainerx_tests/unit_tests/routines_tests/test_math.py
+++ b/tests/chainerx_tests/unit_tests/routines_tests/test_math.py
@@ -1942,6 +1942,8 @@ class TestTan(UnaryMathTestBase, op_utils.NumpyOpTest):
 ))
 class TestAbs(UnaryMathTestBase, op_utils.NumpyOpTest):
 
+    dodge_nondifferentiable = True
+
     def func(self, xp, a):
         assert chainerx.abs is chainerx.absolute
         return xp.abs(a)

